File: src/backend/app/app/models/comment.py
from pydantic import BaseModel, Field
from datetime import datetime

class BaseComment(BaseModel):
    body: str
    rating: int = Field(ge=1, le=5)


class CommentOnCreate(BaseComment):
    id_city: int
    price_per_month: float
    internet_connection: int = Field(ge=1, le=5)
    coworking_spaces: int = Field(ge=1, le=5)
    health_service: int = Field(ge=1, le=5)
    safety: int = Field(ge=1, le=5)
    gastronomy: int = Field(ge=1, le=5)
    means_of_trasnsport: int = Field(ge=1, le=5)
    foreign_friendly: int = Field(ge=1, le=5)
    stay_length: int


class CommentOut(BaseComment):
    id: int
    username: str
    date: datetime
    stay_length: int
    

File: src/frontend/src/utils/client/comment.js
import { CLIENT_BACKEND_BASE_URL } from "./config";

export async function createCommentOnCity(userToken, idCity, comment, ratings) {
    try {
        const result = await fetch(`${CLIENT_BACKEND_BASE_URL}/comment`, {
            method: 'POST',
            headers: {
                'Authorization': `Bearer ${userToken}`,
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({
                "id_city": idCity,
                "body": comment.comment,
                "stay_length": comment.stayLength,
                "price_per_month": comment.pricePerMonth,
                "rating": comment.avgRating,
                "internet_connection": ratings.internetConnection,
                "coworking_spaces": ratings.coworkingSpaces,
                "health_service": ratings.healthService,
                "safety": ratings.safety,
                "gastronomy": ratings.gastronomy,
                "foreign_friendly": ratings.foreignFriendly,
                "means_of_transport": ratings.meansOfTransport,
            })
        });

        const response = await result.json();

        return {
            status: result.status,
            data: result.status === 200 ? response : response.detail
        }
    } catch (error) {
        console.log(error);
    }
}